### ISSUE TITLE
pyparsing-3.1.0 breaks idf.py build (IDFGH-10502)

### ISSUE THREAD

**Issue by morgana2313** [espressif/esp-idf#11748]
### Answers checklist.

- [X] I have read the documentation [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/) and the issue is not addressed there.
- [X] I have updated my IDF branch (master or release) to the latest version and checked that the issue is present there.
- [X] I have searched the issue tracker for a similar issue and not found a similar issue.

### IDF version.

v5.2-dev-1183-gd2fb7240a7

### Operating System used.

Linux

### How did you build your project?

Command line with idf.py

### If you are using Windows, please specify command line type.

None

### What is the expected behavior?

`pyparsing-3.1.0` (released last week) breaks `idf.py build`.  `pyparsing-3.0.9` works without problems.

I expect `~/esp-idf/examples/get-started/hello_world$ idf.py clean &&  idf.py build` to build without errors.

### What is the actual behavior?

Instead it gives a build error:

```
Traceback (most recent call last):
  File "/home/s/esp-idf/tools/ldgen/ldgen/entity.py", line 152, in _get_infos_from_file
    results = parser.parseString(info.content, parseAll=True)
  File "/home/s/.espressif/python_env/idf5.2_py3.10_env/lib/python3.10/site-packages/pyparsing/util.py", line 256, in _inner
    return fn(self, *args, **kwargs)
  File "/home/s/.espressif/python_env/idf5.2_py3.10_env/lib/python3.10/site-packages/pyparsing/core.py", line 1190, in parse_string
    raise exc.with_traceback(None)
pyparsing.exceptions.ParseException: Expected end of text, found 'default'  (at char 1), (line:2, col:1)
```

### Steps to reproduce.

1. Either:
   * Install esp-idf: `git clone --recursive https://github.com/espressif/esp-idf.git && cd esp-idf && ./install.sh` 
   * `~/.espressif/python_env/idf5.2_py3.10_env/bin/pip install --force-reinstall pyparsing==3.1.0`
2. `idf.py build` gives an error
3. `~/.espressif/python_env/idf5.2_py3.10_env/bin/pip install --force-reinstall pyparsing==3.0.9`
4. `idf.py build` builds without error.

### Build or installation Logs.

```plain
s@dev-s5:~/esp-idf/examples/get-started/hello_world$ idf.py clean &&  idf.py build
/home/s/esp-idf/tools/check_python_dependencies.py:12: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
  import pkg_resources
Executing action: clean
Running ninja in directory /home/s/esp-idf/examples/get-started/hello_world/build
Executing "ninja clean"...
[2/2] Cleaning all built files...
Cleaning... 891 files.
Done
/home/s/esp-idf/tools/check_python_dependencies.py:12: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
  import pkg_resources
Executing action: all (aliases: build)
Running ninja in directory /home/s/esp-idf/examples/get-started/hello_world/build
Executing "ninja all"...
[0/1] Re-running CMake...
-- Building ESP-IDF components for target esp32
-- Project sdkconfig file /home/s/esp-idf/examples/get-started/hello_world/sdkconfig
-- Compiler supported targets: xtensa-esp32-elf
-- App "hello_world" version: v5.2-dev-1183-gd2fb7240a7
-- Adding linker script /home/s/esp-idf/examples/get-started/hello_world/build/esp-idf/esp_system/ld/memory.ld
-- Adding linker script /home/s/esp-idf/components/esp_system/ld/esp32/sections.ld.in
-- Adding linker script /home/s/esp-idf/components/esp_rom/esp32/ld/esp32.rom.ld
-- Adding linker script /home/s/esp-idf/components/esp_rom/esp32/ld/esp32.rom.api.ld
-- Adding linker script /home/s/esp-idf/components/esp_rom/esp32/ld/esp32.rom.libgcc.ld
-- Adding linker script /home/s/esp-idf/components/esp_rom/esp32/ld/esp32.rom.newlib-data.ld
-- Adding linker script /home/s/esp-idf/components/esp_rom/esp32/ld/esp32.rom.syscalls.ld
-- Adding linker script /home/s/esp-idf/components/esp_rom/esp32/ld/esp32.rom.newlib-funcs.ld
-- Adding linker script /home/s/esp-idf/components/soc/esp32/ld/esp32.peripherals.ld
-- Components: app_trace app_update bootloader bootloader_support bt cmock console cxx driver efuse esp-tls esp_adc esp_app_format esp_bootloader_format esp_coex esp_common esp_eth esp_event esp_gdbstub esp_hid esp_http_client esp_http_server esp_https_ota esp_https_server esp_hw_support esp_lcd esp_local_ctrl esp_mm esp_netif esp_netif_stack esp_partition esp_phy esp_pm esp_psram esp_ringbuf esp_rom esp_system esp_timer esp_wifi espcoredump esptool_py fatfs freertos hal heap http_parser idf_test ieee802154 json log lwip main mbedtls mqtt newlib nvs_flash nvs_sec_provider openthread partition_table perfmon protobuf-c protocomm pthread sdmmc soc spi_flash spiffs tcp_transport ulp unity usb vfs wear_levelling wifi_provisioning wpa_supplicant xtensa
-- Component paths: /home/s/esp-idf/components/app_trace /home/s/esp-idf/components/app_update /home/s/esp-idf/components/bootloader /home/s/esp-idf/components/bootloader_support /home/s/esp-idf/components/bt /home/s/esp-idf/components/cmock /home/s/esp-idf/components/console /home/s/esp-idf/components/cxx /home/s/esp-idf/components/driver /home/s/esp-idf/components/efuse /home/s/esp-idf/components/esp-tls /home/s/esp-idf/components/esp_adc /home/s/esp-idf/components/esp_app_format /home/s/esp-idf/components/esp_bootloader_format /home/s/esp-idf/components/esp_coex /home/s/esp-idf/components/esp_common /home/s/esp-idf/components/esp_eth /home/s/esp-idf/components/esp_event /home/s/esp-idf/components/esp_gdbstub /home/s/esp-idf/components/esp_hid /home/s/esp-idf/components/esp_http_client /home/s/esp-idf/components/esp_http_server /home/s/esp-idf/components/esp_https_ota /home/s/esp-idf/components/esp_https_server /home/s/esp-idf/components/esp_hw_support /home/s/esp-idf/components/esp_lcd /home/s/esp-idf/components/esp_local_ctrl /home/s/esp-idf/components/esp_mm /home/s/esp-idf/components/esp_netif /home/s/esp-idf/components/esp_netif_stack /home/s/esp-idf/components/esp_partition /home/s/esp-idf/components/esp_phy /home/s/esp-idf/components/esp_pm /home/s/esp-idf/components/esp_psram /home/s/esp-idf/components/esp_ringbuf /home/s/esp-idf/components/esp_rom /home/s/esp-idf/components/esp_system /home/s/esp-idf/components/esp_timer /home/s/esp-idf/components/esp_wifi /home/s/esp-idf/components/espcoredump /home/s/esp-idf/components/esptool_py /home/s/esp-idf/components/fatfs /home/s/esp-idf/components/freertos /home/s/esp-idf/components/hal /home/s/esp-idf/components/heap /home/s/esp-idf/components/http_parser /home/s/esp-idf/components/idf_test /home/s/esp-idf/components/ieee802154 /home/s/esp-idf/components/json /home/s/esp-idf/components/log /home/s/esp-idf/components/lwip /home/s/esp-idf/examples/get-started/hello_world/main /home/s/esp-idf/components/mbedtls /home/s/esp-idf/components/mqtt /home/s/esp-idf/components/newlib /home/s/esp-idf/components/nvs_flash /home/s/esp-idf/components/nvs_sec_provider /home/s/esp-idf/components/openthread /home/s/esp-idf/components/partition_table /home/s/esp-idf/components/perfmon /home/s/esp-idf/components/protobuf-c /home/s/esp-idf/components/protocomm /home/s/esp-idf/components/pthread /home/s/esp-idf/components/sdmmc /home/s/esp-idf/components/soc /home/s/esp-idf/components/spi_flash /home/s/esp-idf/components/spiffs /home/s/esp-idf/components/tcp_transport /home/s/esp-idf/components/ulp /home/s/esp-idf/components/unity /home/s/esp-idf/components/usb /home/s/esp-idf/components/vfs /home/s/esp-idf/components/wear_levelling /home/s/esp-idf/components/wifi_provisioning /home/s/esp-idf/components/wpa_supplicant /home/s/esp-idf/components/xtensa
-- Configuring done
-- Generating done
-- Build files have been written to: /home/s/esp-idf/examples/get-started/hello_world/build
[20/895] Generating ../../partition_table/partition-table.bin
Partition table binary generated. Contents:
*******************************************************************************
# ESP-IDF Partition Table
# Name, Type, SubType, Offset, Size, Flags
nvs,data,nvs,0x9000,24K,
phy_init,data,phy,0xf000,4K,
factory,app,factory,0x10000,1M,
*******************************************************************************
[646/895] Performing configure step for 'bootloader'
-- Building ESP-IDF components for target esp32
-- Project sdkconfig file /home/s/esp-idf/examples/get-started/hello_world/sdkconfig
-- Compiler supported targets: xtensa-esp32-elf
-- Adding linker script /home/s/esp-idf/components/soc/esp32/ld/esp32.peripherals.ld
-- Bootloader project name: "bootloader" version: 1
-- Adding linker script /home/s/esp-idf/components/esp_rom/esp32/ld/esp32.rom.ld
-- Adding linker script /home/s/esp-idf/components/esp_rom/esp32/ld/esp32.rom.api.ld
-- Adding linker script /home/s/esp-idf/components/esp_rom/esp32/ld/esp32.rom.libgcc.ld
-- Adding linker script /home/s/esp-idf/components/esp_rom/esp32/ld/esp32.rom.newlib-funcs.ld
-- Adding linker script /home/s/esp-idf/components/bootloader/subproject/main/ld/esp32/bootloader.ld
-- Adding linker script /home/s/esp-idf/components/bootloader/subproject/main/ld/esp32/bootloader.rom.ld
-- Components: bootloader bootloader_support efuse esp_app_format esp_bootloader_format esp_common esp_hw_support esp_rom esp_system esptool_py freertos hal log main micro-ecc newlib partition_table soc spi_flash xtensa
-- Component paths: /home/s/esp-idf/components/bootloader /home/s/esp-idf/components/bootloader_support /home/s/esp-idf/components/efuse /home/s/esp-idf/components/esp_app_format /home/s/esp-idf/components/esp_bootloader_format /home/s/esp-idf/components/esp_common /home/s/esp-idf/components/esp_hw_support /home/s/esp-idf/components/esp_rom /home/s/esp-idf/components/esp_system /home/s/esp-idf/components/esptool_py /home/s/esp-idf/components/freertos /home/s/esp-idf/components/hal /home/s/esp-idf/components/log /home/s/esp-idf/components/bootloader/subproject/main /home/s/esp-idf/components/bootloader/subproject/components/micro-ecc /home/s/esp-idf/components/newlib /home/s/esp-idf/components/partition_table /home/s/esp-idf/components/soc /home/s/esp-idf/components/spi_flash /home/s/esp-idf/components/xtensa
-- Configuring done
-- Generating done
-- Build files have been written to: /home/s/esp-idf/examples/get-started/hello_world/build/bootloader
[2/3] Generating binary image from built executable
esptool.py v4.6.2
Creating esp32 image...
Merged 2 ELF sections
Successfully created esp32 image.
Generated /home/s/esp-idf/examples/get-started/hello_world/build/bootloader/bootloader.bin
[3/3] cd /home/s/esp-idf/examples/get-started/hello_world/build/bootloader...s/esp-idf/examples/get-started/hello_world/build/bootloader/bootloader.bin
Bootloader binary size 0x68a0 bytes. 0x760 bytes (7%) free.
[891/895] Generating ld/sections.ld
FAILED: esp-idf/esp_system/ld/sections.ld /home/s/esp-idf/examples/get-started/hello_world/build/esp-idf/esp_system/ld/sections.ld
cd /home/s/esp-idf/examples/get-started/hello_world/build/esp-idf/esp_system && /home/s/.espressif/python_env/idf5.2_py3.10_env/bin/python /home/s/esp-idf/tools/ldgen/ldgen.py --config /home/s/esp-idf/examples/get-started/hello_world/sdkconfig --fragments-list "/home/s/esp-idf/components/xtensa/linker.lf;/home/s/esp-idf/components/esp_ringbuf/linker.lf;/home/s/esp-idf/components/driver/linker.lf;/home/s/esp-idf/components/esp_pm/linker.lf;/home/s/esp-idf/components/esp_mm/linker.lf;/home/s/esp-idf/components/spi_flash/linker.lf;/home/s/esp-idf/components/esp_system/linker.lf;/home/s/esp-idf/components/esp_system/app.lf;/home/s/esp-idf/components/esp_rom/linker.lf;/home/s/esp-idf/components/hal/linker.lf;/home/s/esp-idf/components/log/linker.lf;/home/s/esp-idf/components/heap/linker.lf;/home/s/esp-idf/components/soc/linker.lf;/home/s/esp-idf/components/esp_hw_support/linker.lf;/home/s/esp-idf/components/freertos/linker_common.lf;/home/s/esp-idf/components/freertos/linker.lf;/home/s/esp-idf/components/newlib/newlib.lf;/home/s/esp-idf/components/newlib/system_libs.lf;/home/s/esp-idf/components/esp_common/common.lf;/home/s/esp-idf/components/esp_common/soc.lf;/home/s/esp-idf/components/app_trace/linker.lf;/home/s/esp-idf/components/esp_event/linker.lf;/home/s/esp-idf/components/esp_phy/linker.lf;/home/s/esp-idf/components/lwip/linker.lf;/home/s/esp-idf/components/esp_netif/linker.lf;/home/s/esp-idf/components/esp_wifi/linker.lf;/home/s/esp-idf/components/bt/linker.lf;/home/s/esp-idf/components/esp_adc/linker.lf;/home/s/esp-idf/components/esp_gdbstub/linker.lf;/home/s/esp-idf/components/esp_psram/linker.lf;/home/s/esp-idf/components/esp_lcd/linker.lf;/home/s/esp-idf/components/espcoredump/linker.lf;/home/s/esp-idf/components/ieee802154/linker.lf;/home/s/esp-idf/components/openthread/linker.lf" --input /home/s/esp-idf/components/esp_system/ld/esp32/sections.ld.in --output /home/s/esp-idf/examples/get-started/hello_world/build/esp-idf/esp_system/ld/sections.ld --kconfig /home/s/esp-idf/Kconfig --env-file /home/s/esp-idf/examples/get-started/hello_world/build/config.env --libraries-file /home/s/esp-idf/examples/get-started/hello_world/build/ldgen_libraries --objdump /home/s/.espressif/tools/xtensa-esp32-elf/esp-12.2.0_20230208/xtensa-esp32-elf/bin/xtensa-esp32-elf-objdump
Traceback (most recent call last):
  File "/home/s/esp-idf/tools/ldgen/ldgen/entity.py", line 152, in _get_infos_from_file
    results = parser.parseString(info.content, parseAll=True)
  File "/home/s/.espressif/python_env/idf5.2_py3.10_env/lib/python3.10/site-packages/pyparsing/util.py", line 256, in _inner
    return fn(self, *args, **kwargs)
  File "/home/s/.espressif/python_env/idf5.2_py3.10_env/lib/python3.10/site-packages/pyparsing/core.py", line 1190, in parse_string
    raise exc.with_traceback(None)
pyparsing.exceptions.ParseException: Expected end of text, found 'default'  (at char 1), (line:2, col:1)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/s/esp-idf/tools/ldgen/ldgen.py", line 175, in <module>
    main()
  File "/home/s/esp-idf/tools/ldgen/ldgen.py", line 151, in main
    mapping_rules = generation_model.generate(sections_infos)
  File "/home/s/esp-idf/tools/ldgen/ldgen/generation.py", line 514, in generate
    root_node.insert(entity, sections, target, flags, entities)
  File "/home/s/esp-idf/tools/ldgen/ldgen/generation.py", line 241, in insert
    self.child_placement(entity, sections, target, flags, sections_db)
  File "/home/s/esp-idf/tools/ldgen/ldgen/generation.py", line 233, in child_placement
    child.insert(entity, sections, target, flags, sections_db)
  File "/home/s/esp-idf/tools/ldgen/ldgen/generation.py", line 241, in insert
    self.child_placement(entity, sections, target, flags, sections_db)
  File "/home/s/esp-idf/tools/ldgen/ldgen/generation.py", line 233, in child_placement
    child.insert(entity, sections, target, flags, sections_db)
  File "/home/s/esp-idf/tools/ldgen/ldgen/generation.py", line 241, in insert
    self.child_placement(entity, sections, target, flags, sections_db)
  File "/home/s/esp-idf/tools/ldgen/ldgen/generation.py", line 296, in child_placement
    found_sections = sections_db.get_sections(self.parent.name, self.name)
  File "/home/s/esp-idf/tools/ldgen/ldgen/entity.py", line 197, in get_sections
    obj = self._match_obj(archive, obj)
  File "/home/s/esp-idf/tools/ldgen/ldgen/entity.py", line 183, in _match_obj
    objs = self.get_objects(archive)
  File "/home/s/esp-idf/tools/ldgen/ldgen/entity.py", line 176, in get_objects
    self._process_archive(archive)
  File "/home/s/esp-idf/tools/ldgen/ldgen/entity.py", line 164, in _process_archive
    parsed = self._get_infos_from_file(stored)
  File "/home/s/esp-idf/tools/ldgen/ldgen/entity.py", line 154, in _get_infos_from_file
    raise ParseException('Unable to parse section info file ' + info.filename + '. ' + p.msg)
pyparsing.exceptions.ParseException: Unable to parse section info file /home/s/esp-idf/examples/get-started/hello_world/build/esp-idf/esp_event/libesp_event.a. Expected end of text  (at char 0), (line:1, col:1)
ninja: build stopped: subcommand failed.
ninja failed with exit code 1, output of the command is in the /home/s/esp-idf/examples/get-started/hello_world/build/log/idf_py_stderr_output_23717 and /home/s/esp-idf/examples/get-started/hello_world/build/log/idf_py_stdout_output_23717
```

**Comment by igrr** [comment]
https://github.com/espressif/esp-idf/issues/11746

**Comment by dobairoland** [comment]
Anyone who stumbles upon this, there are two related fixes:

1. To re-run the install script again. It should install `pyparsing<3.1` for now.
2. https://github.com/espressif/esp-idf/commit/4020b0deb036ffa3fd01d2dac2297670f65fb7c3 should be the proper fix for `pyparsing>=3.1`.  

The restriction of `pyparsing<3.1` will be lifter later giving enough time to everyone to update to the linked fix.

We are sorry for the inconvenience!
